main.tf

```diff
@@ -110,18 +114,18 @@ resource "aws_default_security_group" "this" {
 }
 
 resource "aws_eip" "nat" {
-  count    = var.enable_nat_gateway && var.nat_as_ec2_instance ? local.ec2_nat_count : 0
+  count    = var.create_vpc && var.enable_nat_gateway && var.nat_as_ec2_instance ? local.ec2_nat_count : 0
   instance = element(aws_instance.nat.*.id, count.index)
   vpc      = true
 }
 
 resource "aws_instance" "nat" {
-  count = var.enable_nat_gateway && var.nat_as_ec2_instance ? local.ec2_nat_count : 0
+  count = var.create_vpc && var.enable_nat_gateway && var.nat_as_ec2_instance ? local.ec2_nat_count : 0
 
   ami                    = data.aws_ami.nat.id
   instance_type          = var.instance_type
   availability_zone      = element(var.availability_zones, count.index)
-  vpc_security_group_ids = [aws_default_security_group.this.id]
+  vpc_security_group_ids = [aws_default_security_group.this[count.index].id]
```

Comment:
looks like there will be an error in case if `local.ec2_nat_count` > 1

Reply:
vpc_security_group_ids = [aws_default_security_group.this[0].id] would work as there is only one default security group in a VPC

Reply:
fixed in commit https://github.com/lean-delivery/tf-module-aws-core/pull/11/commits/7ffc8c85e47c23463014dc897cb79f88d2881894
**[aws_default_security_group.this[0].id] -> aws_default_security_group.this[*].id**